postgres-types/src/lib.rs

```diff
@@ -715,6 +716,31 @@ pub trait ToSql: fmt::Debug {
         ty: &Type,
         out: &mut BytesMut,
     ) -> Result<IsNull, Box<dyn Error + Sync + Send>>;
+
+    /// Specifies the message format of the value
+    fn format(&self) -> Format {
```

Comment:
I don't know how I feel about exposing this on ToSql, since the same can't really be done for FromSql.

An alternative I've been thinking about is to have a separate query method that operates entirely in "string mode", both for parameters and the resulting rows.

Reply:
That's a really good point about the lack of equivalent handle on the returned types. It would be nice to let both parameters and returned types have independently-configurable formats like Postgres itself, but a text-only query variant (`query_text`, perhaps?) would probably cover those cases well-enough.

If we wanted to keep the configurability of each input and output's format, what about adding that configuration to `Statement` instead? Or tracking the configuration of those formats in a `Statement`, generated through `prepare_typed` or some other `prepare_*` method?

Reply:
> An alternative I've been thinking about is to have a separate query method that operates entirely in "string mode", both for parameters and the resulting rows.

@sfackler This would be wonderful; is it anything you've thought more about? Passing arbitrary types as strings is particularly useful, since I could pass any number as a string without knowing the exact intended type and Postgres itself can figure it out. Being able to then load those numbers as strings would also be wonderful. My use case might not be typical, though, so I might be the only person wanting this.